addons/mail/models/discuss/discuss_channel.py

```diff
@@ -721,6 +721,12 @@ def _notify_thread(self, message, msg_vals=False, **kwargs):
         self._bus_send("discuss.channel/new_message", payload)
         return rdata
 
+    def _notify_thread_by_email(self, message, recipients_data, *args, **kwargs):
+        """ Override to prevent email notifications for discuss channels. """
+        if self._name != "discuss.channel":
+            return super()._notify_thread_by_email(message, recipients_data, *args, **kwargs)
+        return True
```

Comment:
- self recordset will never be different than discuss.channel when overridden in the channel model
- we only want to prevent sending the mention emails, not all possible emails (especially in regards to future planned tasks)
- we should probably replace this by sending on inbox instead